apps/zk-explorer/containers/ProjectDetailTabsContainer/Circuits/client/index.tsx

```diff
@@ -0,0 +1,27 @@
+'use client';
+
+import { type FC, useState } from 'react';
+
+import Header from './Header';
+import NavSideBar from './NavSideBar';
+
+import type {
+  GetProjectCircuitDataReturnType,
+  CircuitItemFileType,
+} from '../../../../server';
+
+const CircuitsClient: FC<{ data: GetProjectCircuitDataReturnType }> = ({
+  data,
+}) => {
+  const [activeFile, setActiveFile] = useState<CircuitItemFileType>(
```

Comment:
The way `activeFile` state is initialized could be more efficient. Consider memoizing the initial value with `useMemo` to avoid unnecessary recalculation on every render. For example:

```
const activeFileInit = useMemo(() => Object.values(data).find(item => !item.isFolder), [data]);

const [activeFile, setActiveFile] = useState<CircuitItemFileType>(activeFileInit);
```

Also `find` can be used instead of `filter` as it is more efficient.